### PR TITLE
chore: Add OTLP traces to RTS

### PR DESCRIPTION
**Pull request by nidhi-nair**
Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!IMPORTANT]
> 🟣 🟣 🟣 Your tests are running.
> Tests running at: <https://github.com/appsmithorg/appsmith/actions/runs/11273768049>
> Commit: d4e3bfeeeb84a601eac4033360d700a4ad59a79b
> Workflow: `PR Automation test suite`
> Tags: `@tag.Sanity`
> Spec: ``
> <hr>Thu, 10 Oct 2024 12:13:56 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced OpenTelemetry instrumentation for enhanced tracing and monitoring of HTTP requests.
	- Added new dependencies for OpenTelemetry support.

- **Bug Fixes**
	- Implemented error handling for missing environment variables to ensure server stability.

- **Documentation**
	- Updated logging configurations for better traceability and debugging.

- **Chores**
	- Configured server settings, including API base path and timeout settings.

### CODE REVIEW

File: app/client/packages/rts/src/instrumentation.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+} from "@opentelemetry/semantic-conventions/incubating";
+import { ATTR_SERVICE_NAME } from "@opentelemetry/semantic-conventions";
+import { diag, DiagConsoleLogger, DiagLogLevel } from "@opentelemetry/api";
+import { registerInstrumentations } from "@opentelemetry/instrumentation";
+import { HttpInstrumentation } from "@opentelemetry/instrumentation-http";
+import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-proto";
+
+const provider = new NodeTracerProvider({
+  resource: new Resource({
+    [ATTR_DEPLOYMENT_NAME]: `${process.env.APPSMITH_DEPLOYMENT_NAME || "self-hosted"}`,
```
Comment (dvj1988) [inline app/client/packages/rts/src/instrumentation.ts:18]: Would you be adding the value for `APPSMITH_DEPLOYMENT_NAME` in the cloud deployment repo?

File: app/client/packages/rts/src/instrumentation.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    [ATTR_SERVICE_INSTANCE_ID]: `${process.env.NEW_RELIC_METADATA_KUBERNETES_POD_NAME || "appsmith-0"}`,
```
Comment (dvj1988) [inline app/client/packages/rts/src/instrumentation.ts:19]: Could we use the `HOSTNAME` variable instead? I believe its the pod name in Kubernetes and the container ID in Docker.

File: app/client/packages/rts/src/instrumentation.ts (hunk 1 above)
Comment (nidhi-nair) [reply app/client/packages/rts/src/instrumentation.ts:18]: Yes, that's the idea for now. But we might need another mechanism for self-hosted in the future, gotta think about that

File: app/client/yarn.lock [hunk 3]
```diff
@@ -4964,7 +5075,7 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@opentelemetry/semantic-conventions@npm:1.27.0":
+"@opentelemetry/semantic-conventions@npm:1.27.0, @opentelemetry/semantic-conventions@npm:^1.27.0":
```
Comment (nidhi-nair) [inline app/client/yarn.lock:5078]: @dvj1988 can you help me understand what this line implies? I was confused about having to install semantic conventions for using those constants when it was already present in repo

Comment (dvj1988) [reply app/client/yarn.lock:5078]: This package is already a dependency of the package `@opentelemetry/core@npm:^1.26.0` which was added to `app/client/package.json`. When you added a caret to the version in the rts package.json the yarn workspace just added this to resolve it to the already existing package. You can just leave it as is.

**Review by dvj1988 — APPROVED** [review]

File: app/client/packages/rts/src/instrumentation.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    scheduledDelayMillis: 500,
+    // How long the export can run before it is cancelled
+    exportTimeoutMillis: 30000,
+  },
+);
+
+provider.addSpanProcessor(batchSpanProcessor);
+provider.register();
+
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG);
```
Comment (dvj1988) [inline app/client/packages/rts/src/instrumentation.ts:53]: @nidhi-nair I missed this line. Can you remove this?

**Pull request merged**